OSDOCS#6911: Dual-stack configuration for OpenStack (#63935)
openshift/openshift-docs: modules/install-osp-dualstack.adoc

// Module included in the following assemblies:
//
// * installing/installing_openstack/installing-openstack-installer-custom.adoc
:_content-type: CONCEPT
[id="install-osp-dualstack_{context}]
= Optional: Configuring a cluster with dual-stack networking

:FeatureName: Dual-stack configuration for OpenStack
include::snippets/technology-preview.adoc[]

You can create a dual-stack cluster on {rh-openstack}. However, this is enabled only if you are using an {rh-openstack} network with IPv4 and IPv6 subnets. Before creating a dual-stack cluster, you must complete the following tasks:

* Creating API and Ingress VIPs ports, and specifying the addresses in the `install-config.yaml`.
* Adding the IPv6 subnet to a neutron router to enable router advertisements.

> * Ensuring the dualstack network MTU accommodates both the minimum MTU for IPv6, which is 1280, and the OVN-Kubernetes encapsulation overhead, which is 100.

Reviewer: also, providing dual-stack network in the install-config

Author: Isn't this a part of deployment process?
This section lists 'prerequisites' for deployment, which means, things that has to be in place before we configure dual stack network.
Since the whole topic talks about only configuring dual stack network, the final step is updating install-config. Hence creating network and subnets become prereq along with MTU and other considerations.
Let me know your thoughts here.

Reviewer: To achieve dual-stack OCP cluster you must have the following available before specifying the resources IDs on install-config and triggering installation:

- network created with ipv4 and ipv6 subnets and appropriate mtu
- subnet v6 added to the router
- api and ingress ports created

Those requirements are specific to dual-stack.

Author: @MaysaMacedo can you give this another look? I have modified as per your suggestions

[NOTE]
====

> {rh-openstack} does not support conversion of an IPv4 single-stack cluster to a dual-stack cluster network.

Reviewer: also does not support IPv6 primary dual-stack clusters

// Module included in the following assemblies:
//
// * installing/installing_openstack/installing-openstack-installer-custom.adoc
:_content-type: CONCEPT
[id="install-osp-dualstack_{context}"]
= Optional: Configuring a cluster with dual-stack networking

:FeatureName: Dual-stack configuration for OpenStack
include::snippets/technology-preview.adoc[]

> You can create a dual-stack cluster on {rh-openstack}. However, this is enabled only if you are using an {rh-openstack} network with IPv4 and IPv6 subnets. To configure a dual-stack cluster, you must complete the following tasks:

Reviewer: ```suggestion
You can create a dual-stack cluster on {rh-openstack}. However, this is enabled only if you are using an {rh-openstack} network with IPv4 and IPv6 subnets. Prior to installing a dual-stack cluster, you must complete the following tasks:
```

> You can create a dual-stack cluster on {rh-openstack}. However, this is enabled only if you are using an {rh-openstack} network with IPv4 and IPv6 subnets. To configure a dual-stack cluster, you must complete the following tasks:

Reviewer 2: ```suggestion
You can create a dual-stack cluster on {rh-openstack}. However, this [what?] is enabled only if you are using an {rh-openstack} network with IPv4 and IPv6 subnets. 

To configure a dual-stack cluster, you must complete the following tasks:
```
- The pronoun `this` needs to have a clear reference -- 'this' what? See the ISG for more info: https://www.ibm.com/docs/en/ibm-style?topic=grammar-pronouns#ambiguous-pronoun-references
- Consider adding a paragraph break if you keep this all as a single concept module. I think I would probably break out this sentence and the following content into a separate procedure module.

> 

Reviewer 2: Change all introductory phrases in list items to the imperative voice (I've suggested the changes below) rather than using gerund phrases in sentence fragments. The phrase `you must do the following tasks:` that introduces the list implies that a set of actions that the user must take will follow, and changing to the imperative creates a clear set of actions.

> * Creating a network with IPv4 and IPv6 subnets. The available address modes for `ipv6-ra-mode` and `ipv6-address-mode` fields are: `stateful`, `stateless` and `slaac`.

Reviewer 2: ```suggestion
* Create a network with IPv4 and IPv6 subnets. The available address modes for `ipv6-ra-mode` and `ipv6-address-mode` fields are `stateful`, `stateless` and `slaac`.
```

** The dualstack network MTU must accommodate both the minimum MTU for IPv6, which is 1280, and the OVN-Kubernetes encapsulation overhead, which is 100.

> ** DHCP must be enabled on the subnets.

Reviewer 2: Consider breaking out these two subitems into a Note admonition.

> * Creating API and Ingress VIPs ports.

Reviewer 2: ```suggestion
* Create API and Ingress VIPs ports.
```

> * Adding the IPv6 subnet to a neutron router to enable router advertisements.

Reviewer: ```suggestion
* To enable external connectivity through IPv6 and router advertisements, the user needs to add the dual-stack Network as external gateway in the Router.
```

Author: Tweaking this a bit to adhere to writing principles

> * Adding the dual-stack network as an external gateway in the router to enable external connectivity through IPv6 and router advertisements.

Reviewer: ```suggestion
* To enable router advertisements, the IPv6 subnet needs to be added to the router. In case provider network is used, that can be accomplished by adding the network as an external gateway, which also enables external connectivity.
```

Author: @MaysaMacedo does this look good now?

> * Adding the IPv6 subnet to the router to enable router advertisements. If you are using a provider network, you can enable router advertisements by adding the network as an external gateway. This also enables external connectivity.

Reviewer 2: ```suggestion
* Add the IPv6 subnet to the router to enable router advertisements. If you are using a provider network, you can enable router advertisements by adding the network as an external gateway. This [what??] also enables external connectivity.
```
- The pronoun `this` needs to have a clear reference -- 'this' what? See the ISG for more info: https://www.ibm.com/docs/en/ibm-style?topic=grammar-pronouns#ambiguous-pronoun-references

[NOTE]
====

> {rh-openstack} does not support:

Reviewer 2: [Per the ISG](https://www.ibm.com/docs/en/ibm-style?topic=format-lists#lead-in-wording), reword the lead-in wording for a list to be a complete sentence.
